integrations/slack/webhooks/webhook.go

```diff
@@ -108,21 +111,59 @@ func checkRequest(w http.ResponseWriter, r *http.Request, l *zap.Logger, wantCon
 	}
 
 	// Request body.
-	b, err := io.ReadAll(r.Body)
+	body, err := io.ReadAll(r.Body)
 	if err != nil {
 		l.Error("Failed to read inbound HTTP request body",
 			zap.Error(err),
 		)
 		http.Error(w, "Bad Request", http.StatusBadRequest)
 		return nil
 	}
-	signingSecret := os.Getenv(signingSecretEnvVar)
-	if !verifySignature(signingSecret, ts, sig, b) {
-		l.Error("Slack signature verification failed")
+
+	appID, teamID, enterpriseID, err := h.extractIDs(body, wantContentType, l)
+	if err != nil {
+		l.Error("Failed to extract IDs",
+			zap.Error(err),
+		)
+		http.Error(w, "Bad Request", http.StatusBadRequest)
+		return nil
+	}
+
+	// Get signing secret
+	cids, err := h.listConnectionIDs(r.Context(), appID, enterpriseID, teamID)
+	if err != nil {
+		l.Error("Failed to list connection IDs",
+			zap.Error(err),
+		)
+		http.Error(w, "Bad Request", http.StatusBadRequest)
+		return nil
+	}
+	signingSecret := ""
+	for _, cid := range cids {
```

Comment:
No need for loop-and-break, you can safely assume that the first item (if there is one) is correct. Even if there are multiple connections for the same app/enterprise/workspace IDs, they necessarily share the same signing secret (if it exists).

However, please document this subtle point here for posterity.